nflow-engine/src/main/java/io/nflow/engine/internal/dao/MaintenanceDao.java

```diff
@@ -65,12 +66,20 @@ private String getStateColumns() {
     return stateColumns;
   }
 
-  public List<Long> listOldWorkflows(TablePrefix table, DateTime before, int maxWorkflows) {
+  public List<Long> getOldWorkflowIds(TablePrefix table, DateTime before, int maxWorkflows, Set<String> workflowTypes) {
     String sql = sqlVariants.limit("select id from " + table.nameOf("workflow") + " where next_activation is null and "
-        + sqlVariants.dateLtEqDiff("modified", "?") + " order by id asc", maxWorkflows);
+        + sqlVariants.dateLtEqDiff("modified", "?") + getWorkflowTypeCondition(workflowTypes) + " order by id asc",
+        maxWorkflows);
     return jdbc.queryForList(sql, Long.class, sqlVariants.toTimestampObject(before));
   }
 
+  private String getWorkflowTypeCondition(Set<String> workflowTypes) {
+    if (workflowTypes.isEmpty()) {
+      return "";
+    }
+    return " and type in (" + join(workflowTypes, ",") + ")";
```

Comment:
This is allows sql injection. Random text from outside. At least the rest interface should filter out types that are invalid (or just characters that are not in the basic set)

Reply:
Similar problem seems to be in wakeup rest call that eventually calls `WorkflowInstanceDao.addExpectedStatesToQueryAndUpdate`

Reply:
Added config validation, now only supported workflow definition types are allowed